mongodb/transaction_test.go

```diff
@@ -0,0 +1,255 @@
+package mongodb_test
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"testing"
+	"time"
+
+	mim "github.com/ONSdigital/dp-mongodb-in-memory"
+	mongoDriver "github.com/ONSdigital/dp-mongodb/v3/mongodb"
+	"go.mongodb.org/mongo-driver/bson"
+
+	. "github.com/smartystreets/goconvey/convey"
+)
+
+const (
+	db, collection1, collection2 = "test-db", "test-collection-1", "test-collection-2"
+)
+
+// Example of how to use the Connection.RunTransaction() method to perform a series of mongo operations within a transaction
+func ExampleTransaction() {
```

Comment:
Would it be easier to find this example if it lived in the readme rather than in a test file?

Reply:
yes but I have structured it so that the ExampleTransaction uses/references the exampleTransactionFunc() which is also used in the Tests. I wanted to user to be able to see the example running in the tests. But I can 'copy' it to the README if you think it's worthwhile